### PR TITLE
feat: added signup and login page

### PR DESCRIPTION
**Pull request by xii-vi**
### Added Login and sign-up page with authentication.

ezgif com-gif-maker (5)

### CODE REVIEW

File: src/pages/login/login.jsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  localStorage.setItem('userData', JSON.stringify(loginResponse.data.foundUser));
+  authDispatch({ type: "USER_LOGIN" })
+  authDispatch({ type: "USER_TOKEN", payload: loginResponse.data.encodedToken })
+  authDispatch({ type: "USER_DATA", payload: loginResponse.data.foundUser })
+  }catch(err){
+    console.log(err);
+  }
+  finally{
+    navigate("/product-listing")
+  }
```
Comment (haarsh24) [inline src/pages/login/login.jsx:32]: suggestion:
You can define all API calls in a single folder.

**Review by haarsh24 — APPROVED** [review]
Good to merge

File: src/pages/login/login.jsx (hunk 1 above)
Comment (xii-vi) [reply src/pages/login/login.jsx:32]: _Thanks will do that_

**Review by yashpatel1797 — APPROVED** [review]
good to merge.

**Pull request merged**